[jit] Add Python API for ScriptProfile

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57398 [jit] Add Python API for ScriptProfile**
* #58783 Reapply "[jit] Implement ScriptProfile to collect instruction profiles."

Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

zhxchen17 committed:
WIP _ScriptProfile

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zhxchen17 committed:
Update on "WIP _ScriptProfile"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zhxchen17 committed:
addressed suggestions on "WIP _ScriptProfile"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
nits on "WIP _ScriptProfile"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
rebase on "WIP _ScriptProfile"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
Update on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

gmagogsfm commented on test/jit/test_script_profile.py:
```diff
@@ -0,0 +1,109 @@
+        p = torch.jit._ScriptProfile()
+        p.enable()
+        p.disable()
+        self.assertEqual(p.dump_string(), "")
```
OK to leave for follow up PR:

Can we print the header in this case? Otherwise, it might not be obvious to users that dumping is triggered, just the results is empty.

zhxchen17 committed:
rebase on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
fix test on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
Update on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).

zhxchen17 committed:
Update on "[jit] Add Python API for ScriptProfile"


Summary:
torch.jit._ScriptProfile is a Python level API for using the underlying C++
ScriptProfile facilities, and serves as a pretty printer for structured
profiling data. Currently the support will be mostly about printing results
based on source lines as tables.

Test Plan:
python test/test_jit.py TestScriptProfile

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133577](https://our.internmc.facebook.com/intern/diff/D28133577)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133577).
